### DISCUSSION TITLE
Random screenshot plugin

### DISCUSSION THREAD

**Discussion by victordegnbol** [runelite/runelite#16608] [Ideas]
I have an idea for a new Runelite plug-in: it will randomly screenshot your game (but not too often). Why? When you look through your screenshot folder in the future you will see memories of yourself doing all sorts of content, some of which is content that you'd normally not think about screenshotting. Lightweight plug-in and more memories; for me that's a win-win!

**Comment by victordegnbol** [comment]
* Instead of a new plugin, it should just be an addition to the existing screenshot plugin.

**Reply by Noriomusha** [reply]
Taking a look at adding as a checkbox option on the existing screenshot plugin.

**Reply by Nightfirecat** [reply]
Per the label on this suggestion, we would not accept this in the core client.

**Reply by Noriomusha** [reply]
Understood. Thanks for updating.

**Reply by victordegnbol** [reply]
Thank you for commenting on my suggestion. As I am very new to Github, would you care to explain what the label marked "should-be-external" means? Does your comment, @Nightfirecat, mean you cannot offer this at the moment, or does it mean it could be implemented in another way?

**Reply by Felanbird** [reply]
@victordegnbol The label is to indicate to someone who wanted to add this feature, that it should be added via the plugin-hub and not added to any of the main plugins.

**Reply by victordegnbol** [reply]
@Felanbird Thank you for your reply. Does this mean I have to post the suggestion elsewhere, or does external suggestions also belong here?

**Reply by Felanbird** [reply]
The suggestion can be here, if not you would have already been told it doesn't.

**Reply by neeerp** [reply]
Hey, I'm working on making a plugin-hub plugin for this. I've basically already got a working "prototype" for this that takes a screenshot approximately once every 10 game ticks on average: https://github.com/neeerp/runelite-random-screenshot.

I'll polish this up a bit and put up a PR to plugin hub today or tomorrow.

**Reply by victordegnbol** [reply]
Hello neeerp
Thank you for making the plug-in :) Happy to hear you got the proof of concept working :) Let me know when the plug-in is live.
Cheers!

**Reply by neeerp** [reply]
I've put up a PR here: https://github.com/runelite/plugin-hub/pull/4528. 

The way it works currently is that it has a 1 in `n` probability of taking a screenshot every game tick, where `n` is something you can change in the options. By default, `n` is 500 which translates to a screenshot every 5 minutes on average (one game tick = 0.6 seconds).


Once that's merged, try it out and feel free to offer suggestions on improving it. It's pretty bare right now, but I'd be happy to change things!

**Reply by victordegnbol** [reply]
Hello neeerp
Thank you so much for doing this. I hope other people will enjoy using it as much as me :)
Cheers!

**Reply by neeerp** [reply]
Hey, this was merged yesterday! Let me know if there's anything you'd like me to change!

**Reply by victordegnbol** [reply]
Hello neeerp

I would like to say thank you for the plug-in. It is just as I imagined it :)

Cheers!!

**Reply by neeerp** [reply]
Awesome to hear that @victordegnbol :)
